src/main/java/pl/cleankod/exchange/core/usecase/FindAccountAndConvertCurrencyUseCase.java

```diff
@@ -22,22 +26,26 @@ public FindAccountAndConvertCurrencyUseCase(AccountRepository accountRepository,
         this.baseCurrency = baseCurrency;
     }
 
-    public Optional<Account> execute(Account.Id id, Currency targetCurrency) {
+    public Optional<Account> execute(Account.Id id, Currency targetCurrency, String traceId) {
+        logger.info("{} - Executing find account by ID {} and converting to currency {}", traceId, id, targetCurrency);
         return accountRepository.find(id)
-                .map(account -> new Account(account.id(), account.number(), convert(account.balance(), targetCurrency)));
+                .map(account -> new Account(account.id(), account.number(), convert(account.balance(), targetCurrency, traceId)));
     }
 
-    public Optional<Account> execute(Account.Number number, Currency targetCurrency) {
+    public Optional<Account> execute(Account.Number number, Currency targetCurrency, String traceId) {
+        logger.info("{} - Executing find account by number {} and converting to currency {}", traceId, number, targetCurrency);
         return accountRepository.find(number)
-                .map(account -> new Account(account.id(), account.number(), convert(account.balance(), targetCurrency)));
+                .map(account -> new Account(account.id(), account.number(), convert(account.balance(), targetCurrency, traceId)));
     }
 
-    private Money convert(Money money, Currency targetCurrency) {
+    private Money convert(Money money, Currency targetCurrency, String traceId) {
         if (!baseCurrency.equals(targetCurrency)) {
-            return money.convert(currencyConversionService, targetCurrency);
+            logger.debug("{} - Calling service to convert money {} to target currency {}", traceId, money, targetCurrency);
+            return money.convert(currencyConversionService, targetCurrency, traceId);
         }
 
         if (!money.currency().equals(targetCurrency)) {
+            logger.error("{} - Cannot convert money between the same currency {}", traceId, targetCurrency);
```

Comment:
There is an exception thrown right after this log entry. Looking the exception handler you implemented, we'd loose the ability to connect that response and this log entry. How can we improve this?